src/upnp.cpp

```diff
@@ -225,7 +225,7 @@ int upnp::add_mapping(portmap_protocol const p, int const external_port
 		m.external_port = external_port;
 		m.local_port = local_port;
 
-		if (d.service_namespace) update_map(d, mapping_index);
+		if (!d.service_namespace.empty()) update_map(d, mapping_index);
```

Comment:
service_namespace -> std::string
if(string()) - impossible

int main()
{
  std::string name;
  if(name)   {
     }
}

 In function 'int main()': 8:10: error: could not convert 'name' from 'std::string {aka std::basic_string<char>}' to 'bool'